### PR TITLE
fix: problems with formatting func-call in chain access

### PR DESCRIPTION
**Pull request by QuadnucYard**
This PR includes:

- fix: Args of chained calls are no longer conservatively formatted with structure kept.
- fix: If the access chain starts with a func-call, the args of that func-call are not correctly indented.
- ~change: Enter code mode in expressions with parentheses (array, dict, args, ...). This allows for more linebreaks outside codeblocks.~

### CODE REVIEW

**Comment by QuadnucYard** [comment]
We may need a mode with a name difference from `Code` to express that lines are continued within (allowing breaking binary expressions into multiple lines), which is stronger than `Mode::Code`.

This can be further used for binary chains.

**Comment by QuadnucYard** [comment]
We may want something like
```typ
#{
  let a = (
    1111111111111
      + 2222222222222
      + 3333333333333 / 2
      + 111111111111
        * 333
        / 222
        * 44444
        / 999
        * 1111
        / 44444
        * 1111
        / 44444
        * 1111
        / 44444
      - 222222222222
      + 333333333333
      - 444444444444
      + 555555555555
  )
}
```
instead of
```typ
#{
  let a = (
    1111111111111
      + 2222222222222
      + 3333333333333 / 2
      + (
        111111111111
          * 333
          / 222
          * 44444
          / 999
          * 1111
          / 44444
          * 1111
          / 44444
          * 1111
          / 44444
      )
      - 222222222222
      + 333333333333
      - 444444444444
      + 555555555555
  )
}
```

File: tests/snapshots/assets__check_file@cetz-manual.typ-0.snap [hunk 1]
```diff
@@ -81,9 +82,9 @@ Note that draw functions are imported inside the scope of the `canvas` block. Th
     it
       .text
       .slice(
-      1,
-      -1,
-    ),
+        1,
+        -1,
+      ),
   )
```
Comment (Enter-tainer) [inline tests/snapshots/assets__check_file@cetz-manual.typ-0.snap:88]: cool!

File: tests/snapshots/assets__check_file@packages-tidy-new-parser.typ-0.snap [hunk 2]
```diff
@@ -49,7 +49,10 @@ snapshot_kind: text
     is-named,
   ) = {
     if is-named {
-      return split-once(arg, ":").map(str.trim)
+      return split-once(
+        arg,
+        ":",
+      ).map(str.trim)
     } else {
```
Comment (Enter-tainer) [inline tests/snapshots/assets__check_file@packages-tidy-new-parser.typ-0.snap:56]: I wonder why this is not breakable in previous implementation

File: tests/snapshots/assets__check_file@packages-tidy-new-parser.typ-80.snap [hunk 3]
```diff
@@ -257,10 +257,9 @@ snapshot_kind: text
...
       state.unfinished-param.ends-with(",") or state.state == "finished"
     ) {
-      state.params.push((
-        name: args.first(),
-        desc-lines: state.unmatched-description,
-      ))
+      state
+        .params
+        .push((name: args.first(), desc-lines: state.unmatched-description))
       state.unmatched-description = ()
```
Comment (Enter-tainer) [inline tests/snapshots/assets__check_file@packages-tidy-new-parser.typ-80.snap:263]: The previous one looks better. But I think this is out of this PR's scope. We need determine whether to enter chain mode depending on the length of the chain itself(without parameters of the func call)

File: tests/snapshots/assets__check_file@tablex.typ-80.snap [hunk 4]
```diff
@@ -1475,7 +1480,10 @@ snapshot_kind: text
               align_default: auto,
             )
 
-            let width = measure(cell-box, styles).width // + 2*cell_inset // the box already considers inset
+            let width = measure(
+              cell-box,
+              styles,
+            ).width // + 2*cell_inset // the box already considers inset
```
Comment (Enter-tainer) [inline tests/snapshots/assets__check_file@tablex.typ-80.snap:1486]: seems that in previous versions, when the body of the field access is a function call, we cannot correctly format the function call?

**Review by Enter-tainer — COMMENTED** [review]
The snapshots looks nice. 🚀  Let me review the code later

File: src/pretty/func_call.rs [hunk 5]
```diff
@@ -15,14 +15,21 @@ use super::{
 
 impl<'a> PrettyPrinter<'a> {
     pub(super) fn convert_func_call(&'a self, func_call: FuncCall<'a>) -> ArenaDoc<'a> {
+        if self.current_mode().is_code()
+            && func_call.callee().to_untyped().kind() == SyntaxKind::FieldAccess
+        {
+            return self.convert_dot_chain(func_call.to_untyped());
+        }
```
Comment (Enter-tainer) [inline src/pretty/func_call.rs:22]: okay we do the check here

File: src/pretty/func_call.rs [hunk 6]
```diff
@@ -40,7 +47,7 @@ impl<'a> PrettyPrinter<'a> {
     pub(super) fn convert_args(&'a self, args: Args<'a>) -> ArenaDoc<'a> {
         let has_parenthesized_args = has_parenthesized_args(args);
         let parenthesized = if has_parenthesized_args {
-            self.convert_parenthesized_args_as_is(args)
+            self.convert_parenthesized_args(args)
```
Comment (Enter-tainer) [inline src/pretty/func_call.rs:50]: idk why i initially use as-is for convert_args

File: src/pretty/list.rs [hunk 7]
```diff
@@ -186,7 +188,7 @@ impl<'a> ListStylist<'a> {
         } else if node.kind() == SyntaxKind::Comma {
             self.try_attach_comments();
         } else if node.kind() == SyntaxKind::Space {
-            let newline_cnt = node.text().chars().filter(|c| *c == '\n').count();
+            let newline_cnt = node.text().count_linebreaks();
```
Comment (Enter-tainer) [inline src/pretty/list.rs:191]: nice refactor

**Review by Enter-tainer — APPROVED** [review]
LGTM

**Pull request merged**